processing/src/main/java/io/druid/query/metadata/metadata/SegmentAnalysis.java

```diff
@@ -112,36 +112,19 @@ public boolean equals(Object o)
     if (o == null || getClass() != o.getClass()) {
       return false;
     }
-
     SegmentAnalysis that = (SegmentAnalysis) o;
-
-    if (size != that.size) {
-      return false;
-    }
-
-    if (numRows != that.numRows) {
-      return false;
-    }
-
-    if (id != null ? !id.equals(that.id) : that.id != null) {
-      return false;
-    }
-    if (interval != null ? !interval.equals(that.interval) : that.interval != null) {
-      return false;
-    }
-    return !(columns != null ? !columns.equals(that.columns) : that.columns != null);
-
+    return size == that.size &&
+           numRows == that.numRows &&
+           Objects.equals(id, that.id) &&
+           Objects.equals(interval, that.interval) &&
+           Objects.equals(columns, that.columns) &&
+           Objects.equals(aggregators, that.aggregators);
```

Comment:
not all aggregators might implement equals() correctly, we should probably put a comment here saying this is best-effort only.